docs/host_install.md

```diff
@@ -0,0 +1,189 @@
+for use within a secure enclave.  You **also** need a separate, plain
+copy of Tinyscheme to use outside the enclave for contract development.
+
+- Download the Tinyscheme source:
+```bash
+wget https://downloads.sourceforge.net/project/tinyscheme/tinyscheme/tinyscheme-1.41/tinyscheme-1.41.zip -P /tmp
+```
+
+- Extract and compile it:
+```bash
+cd ${PDO_SOURCE_ROOT}
+unzip /tmp/tinyscheme-1.41.zip
+cd tinyscheme-1.41
+make FEATURES='-DUSE_DL=1 -DUSE_PLIST=1'
+```
+
+- Set the `TINY_SCHEME_SRC` environment variable to the directory where
+you built the package (this environment variable will be used in
+the PDO build process so you might consider adding it to your login
+shell script (`~/.bashrc` or similar):
+
+```bash
+export TINY_SCHEME_SRC=${PDO_SOURCE_ROOT}/tinyscheme-1.41
+```
+
+## Install the SGX SDK
+
+Private Data Objects has been tested with version 2.4 of the SGX
+SDK. You can download prebuilt binaries for the SDK and kernel drivers
+from https://01.org/intel-software-guard-extensions/downloads.
+
+The following commands will download and install version 2.4 of the SGX
+SDK. When asked for the installation directory, we suggest that you install
+the SDK into the directory `/opt/intel`.
+
+```bash
+wget https://download.01.org/intel-sgx/linux-2.4/ubuntu18.04-server/sgx_linux_x64_sdk_2.4.100.48163.bin -P /tmp
+sudo /bin/bash /tmp/sgx_linux_x64_sdk_2.4.100.48163.bin
+```
+
+The installer includes a file that sets environment variables to
+configure the SGX SDK installation. For example, if you installed the
+SDK into the directory `/opt/intel`, then the following will set the
+necessary SGX environment variables:
+
+```bash
+source /opt/intel/sgxsdk/environment
+```
+
+You can also build and install the SGX SDK from source. Instructions for
+building from source are available
+[Intel SGX SDK git repository](https://github.com/intel/linux-sgx).
+
+## Build and Install SGX SSL
+
+SGX OpenSSL is a compilation of OpenSSL specifically for use within SGX
+enclaves. We have tested PDO with SGX SSL version 2.4.1.
+
+Detailed instructions for building and installing SGX SSL is available
+from the
+[Intel SGX SSL github repository](https://github.com/intel/intel-sgx-ssl).
+
+Follow these steps to compile and install version 2.4.1:
+
+- Ensure you have the SGX SDK environment variables activated:
+```bash
+source /opt/intel/sgxsdk/environment
+```
+
+- Clone the SGX SSL source:
+```bash
+git clone 'https://github.com/intel/intel-sgx-ssl.git'
+```
+
+- Check out the recommended version (v2.4.1):
+
+```bash
+cd intel-sgx-ssl
+git checkout v2.4.1
+```
+
+- Download the OpenSSL source package that will form the base of this
+SGX SSL install:
+
+```bash
+cd openssl_source
+wget 'https://www.openssl.org/source/openssl-1.1.0j.tar.gz'
+cd ..
+```
+
+- Set the environment variable for hardware or simulation mode. For
+simulation mode use `export SGX_MODE=SIM`. For hardware mode use `export
+SGX_MODE=HW`.
```

Comment:
in HW mode, the make test part does `not` work without driver and aesmd,
so this should be done after.

Reply:
what are you suggesting? we cover all of the hardware preparation issues in the main install document long before you get to this point.

Reply:
not clear to me which document comes first (from a user perspective).
Options are:
* say that, in HW mode, the command requires driver+psw, or
* remove the `make test` part, or
* suggest to do it in SIM mode

Reply:
I think PSW install instructions should be after immediately driver install instructions (afaik PSW install depends on it) and driver is currently last HW-mode instruction.  Other suggestion migth be to make  the set of HW-mode only steps a bit more explicit other than the the small transition paragraph. What about putting the first part in an Overview subjection, add a HW-mode subsection which includes all hw steps as sub-subsections (i.e., one level lower than right now).  That way we also can reference the hw-configuration if ever there is need from another doc?

Reply:
(answering to the first part) that's exactly what Mic did in his last commit!

Reply:
> (answering to the first part) that's exactly what Mic did in his last commit!

i see it is now included in the install code section.  The section header and preceding text though is a bit misleading as we install here not only kernel driver but also system daemon (aesmd).   (note in SGX doc, driver only corresponds to the kernel model and the other one is (still) called PSW). If we keep both kernel driver and PSW together in a single section we might generalize text and heading into something like 'SGX System Support' rather than 'SGX kernel driver' ?     Lumping kernel driver and service together also makes the final note a bit confusing as only one package will have to be re-installend on OS update (the second one might actually barf if you try to re-run it?) and  'restart the driver' might confuse folks. Maybe clearer might be `Note that if you update your Linux kernel, you may need to reinstall  sgx_linux_x64_driver_778dd1f.bin and potentially restart the system services via systemctl restart aesmd.service`

Reply:
i believe you can edit the text? feel free.

Reply:
Done. updating the branch via my local repo and local edit didn't work the last time i tried but using UI seemed to have worked.